### PR TITLE
Delete shipment support

### PR DESCRIPTION
**Pull request by bartoszkopinski**

### CODE REVIEW

File: lib/dhl_api/request_handler.rb [hunk 1]
```diff
@@ -72,6 +72,7 @@ def parse_response raw_response
           DHLApi::Responses::CreateShipment.new(raw_response)
         when :get_piece
           DHLApi::Responses::GetPiece.new(raw_response)
+        when :deleteShipmentOrder
```
Comment (kunks001) [inline lib/dhl_api/request_handler.rb]: `DHLApi::Responses::DeleteShipment.new(raw_response)` should be here?

File: lib/dhl_api/responses/do_manifest.rb [hunk 2]
```diff
@@ -5,9 +5,9 @@ class DoManifest
 
       def initialize(response)
         @body = response.body[:do_manifest_response]
-        @status_code = body[:manifest_state][:status][:status_code]
```
Comment (kunks001) [inline lib/dhl_api/responses/do_manifest.rb]: was this broken before? please add a spec if one's missing

File: lib/xml/delete_shipment.xml.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<soapenv:Body>
```
Comment (kunks001) [inline lib/xml/delete_shipment.xml.erb]: spec missing

File: spec/request_handler_spec.rb [hunk 4]
```diff
@@ -5,6 +5,32 @@
...
+        expect(response.status_message).to eq("Der Webservice wurde ohne Fehler ausgeführt.")
+      end
+    end
+  end
+
+  context "delete_shipment" do
+    it "cancels shipment order" do
+      VCR.use_cassette('delete_shipment') do
+        response = described_class.request(:delete_shipment, tracking_number: "575007012489")
+        expect(response.status_message).to eq("Der Webservice wurde ohne Fehler ausgeführt.")
```
Comment (kunks001) [inline spec/request_handler_spec.rb]: "The webservice has been executed without error".

Lol. Germans.

File: spec/support/helpers/client_helper.rb [hunk 5]
```diff
@@ -7,7 +7,7 @@ def configure_client **opts
     config.basic_auth_password = opts[:basic_auth_password] || ENV['DHL_BASIC_AUTH_PASSWORD']
     config.zt_id = opts[:zt_id] || ENV['DHL_BASIC_AUTH_USERNAME'] || "xxx"
     config.zt_password = opts[:zt_password] || ENV['DHL_BASIC_AUTH_USERNAME'] || "xxx"
-    config.sandbox = true
+    config.sandbox = false
```
Comment (kunks001) [inline spec/support/helpers/client_helper.rb]: Presumably this means we're hitting the live API during testing? DHL charge us for all shipments booked, even if they don't collect them.

Comment (bartoszkopinski) [reply spec/support/helpers/client_helper.rb]: It doesn't hit the API, we use VCR.

Comment (kunks001) [reply spec/support/helpers/client_helper.rb]: ok - but more generally why are we not changing config to not hit the sandbox in tests?

Comment (bartoszkopinski) [reply spec/support/helpers/client_helper.rb]: DHL sandbox doesn't work

**Pull request merged**